manifests/install_systemd_unit.pp

```diff
@@ -2,15 +2,49 @@
 	$unit_options = {},				# Arguments to pass to the systemd [Unit] section of template
 	$service_options = {},			# Arguments to pass to the systemd [Service] section of template
 	$install_options = {},			# Arguments to pass to the systemd [Install] section of template
+	$timer_options = {},			# Arguments to pass to the systemd [Timer] section of the timer template
 	$service_ensure = "running",	# State of the service after installed
-	$service_enable = true			# Enabled state of the service after installed
+	$service_enable = true,			# Enabled state of the service after installed
+	$path = "/etc/systemd/system",	# Path to systemd files
 ) {
 
-	::systemd::unit_file { "${name}.service":
-		content => template("${module_name}/systemd_service.erb")
-	} -> service { $name:
-		ensure => $service_ensure,
-		enable => $service_enable,
-		require => Exec['systemctl-daemon-reload'],
+	if $name =~ /^([A-Za-z0-9_-]+)(\.)(timer)$/ {
+		$real_name = $name
+		$timer_service = regsubst($real_name, '.timer', '.service', 'G')
+		$service = ""
+		$exec = "systemctl start ${real_name}"
+	} elsif $name =~ /^([A-Za-z0-9_-]+)(\.)(service)$/ {
+		$real_name = $name
+		$service = regsubst($real_name, '.service', '', 'G')
+		$exec = ""
+	} else {
+		$real_name = "${name}.service"
+		$service = regsubst($real_name, '.service', '', 'G')
+		$exec = ""
+		notice("${name} does not include '.service' or '.timer', assuming '.service'")
 	}
-}
+
+	::systemd::unit_file { "${real_name}":
+		content => template("${module_name}/systemd_unit.erb"),
+		path => $path,
+	}
+
+	if !empty($service) {
+		service { $service:
+			ensure => $service_ensure,
+			enable => $service_enable,
+			notify => Exec['systemctl-daemon-reload'],
+			require => File["${path}/${real_name}"],
+		}
+	}
+
+	if !empty($exec) {
+		exec { "systemctl_start_${real_name}":
+			command => $exec,
+			require => File["${path}/${real_name}"],
+			subscribe => File["${path}/${real_name}"],
+			refreshonly => false,
```

Comment:
don't think it's needed, it seems to do `systemctl start` but that will get handled by the service

Reply:
A timer isn't a service. The Service resource in puppet assumes you are a .service file. So if you pass it "foo.timer" it will try to start "foo.timer.service"